### ISSUE TITLE
Define and document domain adaptation for the current v2 extractor

### ISSUE THREAD

**Issue by adaamko** [KRLabsOrg/verbatim-rag#30]
## Goal

Define and document a supported domain-adaptation workflow for the **current v2
token/highlighter extractor**.

The previous issue body pointed to the in-repository
`packages/core/verbatim_core/extractor_models/` trainer. That code trains the
legacy sentence-level `QAModel`; it does not train a checkpoint compatible with
the current `KRLabsOrg/verbatim-rag-modern-bert-v2` highlighter path.

The current v2 research training and canonical evaluation code lives in:

- [`KRLabsOrg/acl-verbatim`](https://github.com/KRLabsOrg/acl-verbatim)
- [`KRLabsOrg/verbatim-spans`](https://huggingface.co/datasets/KRLabsOrg/verbatim-spans)

This issue should turn that research workflow into an accurate, versioned path
for a user adapting v2 to a private domain. It must not present the legacy
trainer as v2 fine-tuning.

## Required workflow

The guide should cover:

1. **Annotation schema** — question, source text, and one or more exact
   character-offset spans; include negative rows with no evidence.
2. **Data validation** — offsets in bounds, `text == source[start:end]`, stable
   row ids, and explicit train/validation/test roles.
3. **Conversion** — a reproducible command that converts the documented schema
   into the v2 token-classification format used by the pinned training code.
4. **Training** — exact repository revision, command/config, base model,
   dependencies, seed, hardware expectations, and output artifacts.
5. **Evaluation** — the canonical micro Word-P/R/F1 scorer on a held-out set,
   including negative examples; threshold selection must not use the final test
   set.
6. **Loading** — demonstrate the resulting checkpoint with
   `ModelSpanExtractor(model_path=...)` and verify an exact source slice.
7. **Artifact manifest** — model/data revisions, thresholds, package versions,
   and known limitations.

## Implementation decision

Before writing a polished tutorial, verify whether the current
`acl-verbatim` commands are stable enough to support externally. If not, the
first PR should add the smallest reproducible training entry point there (or a
small supported wrapper here) and then document it. The issue can span the two
repositories, but runtime training logic should have one source of truth.

## Acceptance criteria

- [ ] No path, command, schema, or checkpoint name refers to the legacy trainer
      as if it trained v2.
- [ ] Every command is tested against a pinned `acl-verbatim` revision.
- [ ] A tiny fixture exercises schema validation and conversion without running
      a full model training job.
- [ ] A smoke configuration completes a short training/evaluation cycle in the
      documented environment, with its logs and artifact shape recorded.
- [ ] The resulting checkpoint loads through the public v2
      `ModelSpanExtractor` path.
- [ ] The evaluation protocol separates training, threshold calibration, and
      held-out reporting.
- [ ] The guide clearly distinguishes silver labels from human labels and does
      not promise transfer quality before evaluation on the user's domain.
- [ ] Documentation links the canonical evaluation harness tracked in #29.

## Non-goals

- documenting the legacy `QAModel` trainer as the recommended path;
- bundling a generic training framework into the runtime package;
- downloading models or running full training in normal unit CI;
- claiming that a small synthetic example establishes production quality.

## Start here

```bash
git clone https://github.com/KRLabsOrg/acl-verbatim.git
grep -R "ModernBertForTokenClassification\|Trainer\|span_metrics" -n acl-verbatim
```

**Comment by adaamko** [comment]
Triage correction: this issue is reopened and its body has been updated in
place rather than replaced. The goal remains a custom-domain fine-tuning guide,
but the specification now targets the current v2 token/highlighter model and
the actual `acl-verbatim` training source of truth. The legacy in-package
`QAModel` trainer is documented only to prevent users from training an
incompatible checkpoint by mistake.
